src/main/java/com/example/bookstore/service/book/BookServiceImpl.java

```diff
@@ -66,4 +68,12 @@ public List<BookDto> search(BookSearchParametersDto searchParameters) {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> getBooksByCategoryId(Long categoryId) {
+        List<Book> books = bookRepository.findAllByCategoryId(categoryId);
+        return books.stream()
+                .map(bookMapper::toDtoWithoutCategoryIds)
+                .collect(Collectors.toList());
```

Comment:
You are using JDK 17, so feel free to use simpler approach
```suggestion
                .toList();
```